os-server/src/main/java/app/openschool/user/UserServiceImpl.java

```diff
@@ -106,4 +109,13 @@ public List<UserCourseDto> findUserEnrolledCourses(Long userId, Long courseStatu
     return UserCourseMapper.toUserCourseDtoList(
         enrolledCourseRepository.findUserEnrolledCoursesByStatus(userId, courseStatusId));
   }
+
+  @Override
+  @Transactional
+  public Page<MentorCourseDto> findCoursesByMentorId(Long mentorId, Pageable page) {
+    Optional<User> user = userRepository.findUserById(mentorId);
```

Comment:
You can define a method which is calling userRepository.findUserById method and the returned optional check in the controller, in case of empty you can return response with bad request status code. In this case you don't need to throw exception in the service.

Reply:
changed